src/app/util/resize_image.cpp

```diff
@@ -38,7 +38,9 @@ doc::Image* resize_image(const doc::Image* image,
   std::unique_ptr<doc::Image> newImage(doc::Image::create(spec));
   newImage->setMaskColor(image->maskColor());
 
-  doc::algorithm::fixup_image_transparent_colors(newImage.get());
+  doc::algorithm::fixup_image_transparent_colors(
+    newImage.get(),
+    method == doc::algorithm::RESIZE_METHOD_NEAREST_NEIGHBOR);
```

Comment:
As the argument is used to skip the function call, I think we have some other options:

1. One option would be to skip the call at the function call site:
```suggestion
  if (doc::algorithm::does_resize_method_require_fixup(method))
    doc::algorithm::fixup_image_transparent_colors(newImage.get());
```

2. Other option would be to pass the `method` to the function:
```suggestion
  doc::algorithm::fixup_image_transparent_colors(newImage.get(), method);
```

3. But I was thinking that probably the `fixup_image_transparent_colors` can be called inside `resize_image` when the `method` is not `RESIZE_METHOD_NEAREST_NEIGHBOR`.

Probably the third option is the best one, as the `resize_image()` already requires the fixup for other resize methods.

Reply:
Thanks for the suggestions! However,  the third option requires changing the  first param of `resize_image()` to non-const pointer, which might lead to reduced performance. One workaround is to declare a new const pointer after calling `fixup_image_transparent_colors`, like this
```
void resize_image(Image* src,
                  Image* dst,
                  const ResizeMethod method,
                  const Palette* pal,
                  const RgbMap* rgbmap,
                  const color_t maskColor)
{
  if(RESIZE_METHOD_NEAREST_NEIGHBOR != method)
    doc::algorithm::fixup_image_transparent_colors(src);
  const Image* csrc = src;
  //other code...
}
```
But I`m not sure if there is a better solution. Personally, I think the first option is the simplest and safest way to fix this bug, even though it might not be the perfect one.